pkg/pillar/cmd/msrv/handlers.go

```diff
@@ -787,3 +805,73 @@ func (msrv *Msrv) handleActivateCredentialPost() func(http.ResponseWriter, *http
 		w.Write(out)
 	}
 }
+
+func (msrv *Msrv) handleNetworkStatusMetrics() func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		msrv.Log.Tracef("networkStatusMetricsHandler.ServeHTTP")
+
+		// Fetch network status data
+		devNetStatusObj, err := msrv.subDeviceNetworkStatus.Get("global")
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusNoContent), http.StatusNoContent)
+			return
+		}
+		devNetStatus := devNetStatusObj.(types.DeviceNetworkStatus)
+
+		// Fetch network metrics data
+		metricsObj, err := msrv.subNetworkMetrics.Get("global")
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusNoContent), http.StatusNoContent)
+			return
+		}
+		metrics := metricsObj.(types.NetworkMetrics)
+
+		// List only physical interfaces
```

Comment:
Actually just to clarify (for the documentation), this new endpoint will provide metrics for all (used) device ports. That includes physical interfaces (eth, WiFi, modems) and also virtual interfaces created on top of them - VLANs and LAGs. What is not included are virtual interfaces used for app connectivity (bridges, qemu TAPs, etc.)
Also please mention in docs that for wwan interfaces this endpoint provides Linux counters, while the `wwan/metrics.json` has metrics collected directly from modems. There can be differences if modem is dropping packets or if it is stuck etc.

Reply:
Thanks for the clarification @milan-zededa , I will update the PR accordingly...